Error inflating class com.suke.widget.SwitchButton

FATAL EXCEPTION: main
                                                                          java.lang.RuntimeException: Unable to start activity ComponentInfo{com.improlabs.optima.ucs/com.improlabs.optima.ucs.MainActivity}: android.view.InflateException: Binary XML file line #185: Error inflating class com.suke.widget.SwitchButton
                                                                              at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2059)
                                                                              at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084)
                                                                              at android.app.ActivityThread.access$600(ActivityThread.java:130)
                                                                              at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195)
                                                                              at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                              at android.os.Looper.loop(Looper.java:137)
                                                                              at android.app.ActivityThread.main(ActivityThread.java:4745)
                                                                              at java.lang.reflect.Method.invokeNative(Native Method)
                                                                              at java.lang.reflect.Method.invoke(Method.java:511)
                                                                              at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786)
                                                                              at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553)
                                                                              at dalvik.system.NativeStart.main(Native Method)
                                                                           Caused by: android.view.InflateException: Binary XML file line #185: Error inflating class com.suke.widget.SwitchButton
                                                                              at android.view.LayoutInflater.createView(LayoutInflater.java:613)
                                                                              at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:687)
                                                                              at android.view.LayoutInflater.rInflate(LayoutInflater.java:746)
                                                                              at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                              at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                              at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                              at android.view.LayoutInflater.rInflate(LayoutInflater.java:749)
                                                                              at android.view.LayoutInflater.inflate(LayoutInflater.java:489)
                                                                              at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
                                                                              at com.improlabs.optima.fragments.SettingsFragment.onCreateView(SettingsFragment.java:80)
                                                                              at android.support.v4.app.Fragment.performCreateView(Fragment.java:1974)
                                                                              at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
                                                                              at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1252)
                                                                              at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1234)
                                                                              at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2046)
                                                                              at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:174)
                                                                              at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:597)
                                                                              at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1163)
                                                                              at android.app.Activity.performStart(Activity.java:5018)
                                                                              at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2032)
                                                                              at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2084) 
                                                                              at android.app.ActivityThread.access$600(ActivityThread.java:130) 
                                                                              at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1195) 
                                                                              at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                              at android.os.Looper.loop(Looper.java:137) 
                                                                              at android.app.ActivityThread.main(ActivityThread.java:4745) 
                                                                              at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                              at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                              at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:786) 
                                                                              at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:553) 
                                                                              at dalvik.system.NativeStart.main(Native Method) 

Same problem in my project:
```
03-16 14:12:27.367 30767-30767/com.parkingwang.iop E/AndroidRuntime: FATAL EXCEPTION: main
                                                                     java.lang.RuntimeException: Unable to start activity ComponentInfo{com.parkingwang.iop/com.parkingwang.iop.core.summary.DataSummaryAty}: android.view.InflateException: Binary XML file line #8: Error inflating class com.parkingwang.iop.widgets.uis.LineChart
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2081)
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106)
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:137)
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202)
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                         at android.os.Looper.loop(Looper.java:137)
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4819)
                                                                         at java.lang.reflect.Method.invokeNative(Native Method)
                                                                         at java.lang.reflect.Method.invoke(Method.java:511)
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
                                                                         at dalvik.system.NativeStart.main(Native Method)
                                                                      Caused by: android.view.InflateException: Binary XML file line #8: Error inflating class com.parkingwang.iop.widgets.uis.LineChart
                                                                         at android.view.LayoutInflater.createView(LayoutInflater.java:683)
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:816)
                                                                         at android.view.LayoutInflater.parseInclude(LayoutInflater.java:900)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:806)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:559)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:466)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:417)
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:288)
                                                                         at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
                                                                         at com.parkingwang.iop.base.BaseActivity.setContentView(BaseActivity.java:122)
                                                                         at com.parkingwang.iop.base.BaseActivity.setContentView(BaseActivity.java:118)
                                                                         at com.parkingwang.iop.core.summary.DataSummaryAty.setUpContentView(DataSummaryAty.java:109)
                                                                         at com.parkingwang.iop.base.BaseActivity.onCreate(BaseActivity.java:75)
                                                                         at android.app.Activity.performCreate(Activity.java:5084)
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2045)
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106) 
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:137) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4819) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
                                                                      Caused by: java.lang.reflect.InvocationTargetException
                                                                         at java.lang.reflect.Constructor.constructNative(Native Method)
                                                                         at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
                                                                         at android.view.LayoutInflater.createView(LayoutInflater.java:657)
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:816) 
                                                                         at android.view.LayoutInflater.parseInclude(LayoutInflater.java:900) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:559) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:466) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:417) 
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:288) 
                                                                         at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) 
                                                                         at com.parkingwang.iop.base.BaseActivity.setContentView(BaseActivity.java:122) 
                                                                         at com.parkingwang.iop.base.BaseActivity.setContentView(BaseActivity.java:118) 
                                                                         at com.parkingwang.iop.core.summary.DataSummaryAty.setUpContentView(DataSummaryAty.java:109) 
                                                                         at com.parkingwang.iop.base.BaseActivity.onCreate(BaseActivity.java:75) 
                                                                         at android.app.Activity.performCreate(Activity.java:5084) 
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2045) 
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106) 
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:137) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4819) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
                                                                      Caused by: android.view.InflateException: Binary XML file line #26: Error inflating class com.suke.widget.SwitchButton
                                                                         at android.view.LayoutInflater.createView(LayoutInflater.java:683)
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:816)
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:559)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:466)
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:417)
                                                                         at com.parkingwang.iop.widgets.uis.LineChart.setUpView(LineChart.java:69)
                                                                         at com.parkingwang.iop.widgets.uis.LineChart.<init>(LineChart.java:60)
                                                                         at java.lang.reflect.Constructor.constructNative(Native Method) 
                                                                         at java.lang.reflect.Constructor.newInstance(Constructor.java:417) 
                                                                         at android.view.LayoutInflater.createView(LayoutInflater.java:657) 
                                                                         at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:816) 
                                                                         at android.view.LayoutInflater.parseInclude(LayoutInflater.java:900) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:806) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.rInflate(LayoutInflater.java:819) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:559) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:466) 
                                                                         at android.view.LayoutInflater.inflate(LayoutInflater.java:417) 
                                                                         at android.support.v7.app.AppCompatDelegateImplV9.setContentView(AppCompatDelegateImplV9.java:288) 
                                                                         at android.support.v7.app.AppCompatActivity.setContentView(AppCompatActivity.java:140) 
                                                                         at com.parkingwang.iop.base.BaseActivity.setContentView(BaseActivity.java:122) 
                                                                         at com.parkingwang.iop.base.BaseActivity.setContentView(BaseActivity.java:118) 
                                                                         at com.parkingwang.iop.core.summary.DataSummaryAty.setUpContentView(DataSummaryAty.java:109) 
                                                                         at com.parkingwang.iop.base.BaseActivity.onCreate(BaseActivity.java:75) 
                                                                         at android.app.Activity.performCreate(Activity.java:5084) 
                                                                         at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
                                                                         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2045) 
                                                                         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106) 
                                                                         at android.app.ActivityThread.access$600(ActivityThread.java:137) 
                                                                         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202) 
                                                                         at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                         at android.os.Looper.loop(Looper.java:137) 
                                                                         at android.app.ActivityThread.main(ActivityThread.java:4819) 
                                                                         at java.lang.reflect.Method.invokeNative(Native Method) 
                                                                         at java.lang.reflect.Method.invoke(Method.java:511) 
                                                                         at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
                                                                         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
                                                                         at dalvik.system.NativeStart.main(Native Method) 
                                                                      Caused by: java.lang.reflect.InvocationTargetException
                                                                         at java.lang.reflect.Constructor.constructNative(Native Method)
                                                                         at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
                                                                     	at android.view.LayoutInflater.createView(LayoutInflater.j
```

Android 4.1.2

The crash also happened when your sample project was run on Android 4.1.2.
```
E/AndroidRuntime: FATAL EXCEPTION: main
  java.lang.RuntimeException: Unable to start activity ComponentInfo{com.suke.widget.sample/com.suke.widget.sample.MainActivity}: android.view.InflateException: Binary XML file line #9: Error inflating class com.suke.widget.SwitchButton
  at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2081)
  at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106)
  at android.app.ActivityThread.access$600(ActivityThread.java:137)
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202)
  at android.os.Handler.dispatchMessage(Handler.java:99)
  at android.os.Looper.loop(Looper.java:137)
  at android.app.ActivityThread.main(ActivityThread.java:4819)
  at java.lang.reflect.Method.invokeNative(Native Method)
  at java.lang.reflect.Method.invoke(Method.java:511)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560)
  at dalvik.system.NativeStart.main(Native Method)
  Caused by: android.view.InflateException: Binary XML file line #9: Error inflating class com.suke.widget.SwitchButton
  at android.view.LayoutInflater.createView(LayoutInflater.java:683)
  at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757)
  at android.view.LayoutInflater.rInflate(LayoutInflater.java:816)
  at android.view.LayoutInflater.inflate(LayoutInflater.java:559)
  at android.view.LayoutInflater.inflate(LayoutInflater.java:466)
  at android.view.LayoutInflater.inflate(LayoutInflater.java:417)
  at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:328)
  at android.app.Activity.setContentView(Activity.java:1942)
  at com.suke.widget.sample.MainActivity.onCreate(MainActivity.java:12)
  at android.app.Activity.performCreate(Activity.java:5084)
  at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079)
  at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2045)
  at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106) 
  at android.app.ActivityThread.access$600(ActivityThread.java:137) 
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202) 
  at android.os.Handler.dispatchMessage(Handler.java:99) 
  at android.os.Looper.loop(Looper.java:137) 
  at android.app.ActivityThread.main(ActivityThread.java:4819) 
  at java.lang.reflect.Method.invokeNative(Native Method) 
  at java.lang.reflect.Method.invoke(Method.java:511) 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
  at dalvik.system.NativeStart.main(Native Method) 
  Caused by: java.lang.reflect.InvocationTargetException
  at java.lang.reflect.Constructor.constructNative(Native Method)
  at java.lang.reflect.Constructor.newInstance(Constructor.java:417)
  at android.view.LayoutInflater.createView(LayoutInflater.java:657)
  at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757) 
  at android.view.LayoutInflater.rInflate(LayoutInflater.java:816) 
  at android.view.LayoutInflater.inflate(LayoutInflater.java:559) 
  at android.view.LayoutInflater.inflate(LayoutInflater.java:466) 
  at android.view.LayoutInflater.inflate(LayoutInflater.java:417) 
  at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:328) 
  at android.app.Activity.setContentView(Activity.java:1942) 
  at com.suke.widget.sample.MainActivity.onCreate(MainActivity.java:12) 
  at android.app.Activity.performCreate(Activity.java:5084) 
  at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
  at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2045) 
  at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106) 
  at android.app.ActivityThread.access$600(ActivityThread.java:137) 
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202) 
  at android.os.Handler.dispatchMessage(Handler.java:99) 
  at android.os.Looper.loop(Looper.java:137) 
  at android.app.ActivityThread.main(ActivityThread.java:4819) 
  at java.lang.reflect.Method.invokeNative(Native Method) 
  at java.lang.reflect.Method.invoke(Method.java:511) 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
  at dalvik.system.NativeStart.main(Native Method) 
  Caused by: java.lang.RuntimeException: illegal call : method [setPadding]
  at com.suke.widget.SwitchButton.setPadding(SwitchButton.java:68)
  at android.view.View.<init>(View.java:3741)
  at android.view.View.<init>(View.java:3355)
  at com.suke.widget.SwitchButton.<init>(SwitchButton.java:50)
  at java.lang.reflect.Constructor.constructNative(Native Method) 
  at java.lang.reflect.Constructor.newInstance(Constructor.java:417) 
  at android.view.LayoutInflater.createView(LayoutInflater.java:657) 
  at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:757) 
  at android.view.LayoutInflater.rInflate(LayoutInflater.java:816) 
  at android.view.LayoutInflater.inflate(LayoutInflater.java:559) 
  at android.view.LayoutInflater.inflate(LayoutInflater.java:466) 
  at android.view.LayoutInflater.inflate(LayoutInflater.java:417) 
  at com.android.internal.policy.impl.PhoneWindow.setContentView(PhoneWindow.java:328) 
  at android.app.Activity.setContentView(Activity.java:1942) 
  at com.suke.widget.sample.MainActivity.onCreate(MainActivity.java:12) 
  at android.app.Activity.performCreate(Activity.java:5084) 
  at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1079) 
  at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2045) 
  at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2106) 
  at android.app.ActivityThread.access$600(ActivityThread.java:137) 
  at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1202) 
  at android.os.Handler.dispatchMessage(Handler.java:99) 
  at android.os.Looper.loop(Looper.java:137) 
  at android.app.ActivityThread.main(ActivityThread.java:4819) 
  at java.lang.reflect.Method.invokeNative(Native Method) 
  at java.lang.reflect.Method.invoke(Method.java:511) 
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:793) 
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:560) 
  at dalvik.system.NativeStart.main(Native Method) 
```

--- a/library/src/main/java/com/suke/widget/SwitchButton.java
+++ b/library/src/main/java/com/suke/widget/SwitchButton.java
@@ -64,8 +64,7 @@ public class SwitchButton extends View implements Checkable {
 
     @Override
     public final void setPadding(int left, int top, int right, int bottom) {
-        //do nothing
-        throw new RuntimeException("illegal call : method [setPadding]");
+        super.setPadding(0, 0, 0, 0);
     }
 
     /**
@@ -186,7 +185,7 @@ public class SwitchButton extends View implements Checkable {
         valueAnimator.addListener(animatorListener);
 
         super.setClickable(true);
-        super.setPadding(0,0,0,0);
+        this.setPadding(0, 0, 0, 0);
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.HONEYCOMB) {
             setLayerType(LAYER_TYPE_SOFTWARE, null);
         }

